Code/GraphMol/Descriptors/EEM.cpp

```diff
@@ -0,0 +1,237 @@
+const float A2[] = {0.0,0.0,   0.0,0.0,0.0,0.0,2.7667,2.8895,3.0486,0.0,   0.0,0.0,0.0,0.0,0.0,2.2933,2.6471,   0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+const float A3[] = {0.0,0.0,   0.0,0.0,0.0,0.0,2.6944,3.0240,0.0,   0.0,   0.0,0.0,0.0,0.0,0.0,0.0,   0.0,      0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+                   //0   1      2   3   4   5   6       7      8      9     10  11  12  13  14  15      16    17     18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35
+const float B1[] = {0.0,1.1641,0.0,0.0,0.0,0.0,0.6403,0.9083,1.0577,0.9983,0.0,0.0,0.0,0.0,0.0,0.0,   0.4897,1.1537,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.1105};
+const float B2[] = {0.0,0.0,   0.0,0.0,0.0,0.0,0.6513,0.6647,0.8410,0.0,   0.0,0.0,0.0,0.0,0.0,0.5759,0.4512,   0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+const float B3[] = {0.0,0.0,   0.0,0.0,0.0,0.0,0.6776,1.4240,0.0,   0.0,   0.0,0.0,0.0,0.0,0.0,0.0,   0.0,      0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+
+/* simple Parameters trained using the small 500 molecules dataset.
+const float kappa = 0.1960;
+
+const float A1[] = {0.0,2.3594,0.0,0.0,0.0,0.0,2.4541,2.5908,2.7130,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.3833};
+const float A2[] = {0.0,0.0,0.0,0.0,0.0,0.0,2.4726,2.5409,2.6766,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.4956};
+const float A3[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+
+const float B1[] = {0.0,0.5962,0.0,0.0,0.0,0.0,0.2591,0.3316,0.5028,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.4564};
+const float B2[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.2268,0.2319,0.4992,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.1493};
+const float B3[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+*/
+
+// function to retreive the atomtype value based on the highest bond type of an atom
+// in the publication they don't have access to "Aromatic type"
+double getAtomtype(const RDKit::ROMol mol, const RDKit::Atom *atom) {
+      double t=1.0;
+      double a;
+      RDKit::ROMol::ADJ_ITER nbrIdx, endNbrs;
+      boost::tie(nbrIdx, endNbrs) = mol.getAtomNeighbors(atom);
+      while (nbrIdx != endNbrs) {
+        const RDKit::Bond *bond = mol.getBondBetweenAtoms(*nbrIdx, atom->getIdx());
+        a = bond->getBondTypeAsDouble();
+        if (a == 1.5) {
+             t = 2.0;
+            }
+        if (a == 2.0) {
+             t = 2.0;
+            }
+        if (a == 3.0) {
+             t = 3.0;
+            }
+        ++nbrIdx;
+  }
+  return t;
+}
+
+double* getEEMMatrix(RDKit::ROMol mol, double * dist3D, int n, EEM_arrays EEMatoms) {
+  int sizeArray = (n+1) * (n+1);
+  double* EEM = new double[sizeArray](); // declaration to set all elements to zeros!
+  #define IDX(x, y) (x * (n+1) + y)
```

Comment:
I'm not terribly fond of the style of introducing a macro to do this calculation instead of just having `x * (n+1) +y` explicitly present in the code. This doesn't save you all that much typing and does obfuscate things a bit.